docs/userguide/perturbative_solvers.rst

```diff
@@ -0,0 +1,308 @@
+        dt=dt,
+        carrier_freqs=[v],
+        chebyshev_orders=[1],
+        expansion_order=7,
+        integration_method='jax_odeint',
+        atol=1e-12,
+        rtol=1e-12
+    )
+
+
+Construct a function that simulates the system for the pulse sequence
+with a given amplitude.
+
+.. jupyter-execute::
+
+    from qiskit_dynamics import Signal
+
+    def dyson_sim(amp):
+        drive_signal = Signal(lambda t: Array(amp) * envelope_func(t), carrier_freq=v)
+        return dyson_solver.solve(
+            signals=[drive_signal],
+            y0=np.eye(dim, dtype=complex),
+            t0=0.,
+            n_steps=int(T // dt)
+        ).y[-1]
+
+    from jax import jit
+
+    jit_dyson_sim = jit(dyson_sim)
+
+First run includes compile time.
+
+.. jupyter-execute::
+
+    %time yf_dyson = jit_dyson_sim(1.).block_until_ready()
+
+
+Second run demonstrates the speed of the solver.
+
+.. jupyter-execute::
+
+    %time yf_dyson = jit_dyson_sim(1.).block_until_ready()
+
+
+4. Comparison to traditional ODE solver
+---------------------------------------
+
+We now construct the same simulation using a standard solver to compare
+accuracy and simulation speed.
+
+.. jupyter-execute::
+
+    from qiskit_dynamics import Solver
+
+    solver = Solver(
+        static_hamiltonian=static_hamiltonian,
+        hamiltonian_operators=[drive_hamiltonian],
+        rotating_frame=static_hamiltonian
+    )
+
+    # specify tolerance as an argument to run the simulation at different tolerances
+    def ode_sim(amp, tol):
+        drive_signal = Signal(lambda t: Array(amp) * envelope_func(t), carrier_freq=v)
+        res = solver.solve(
+            t_span=[0., int(T // dt) * dt],
+            y0=np.eye(dim, dtype=complex),
+            signals=[drive_signal],
+            method='jax_odeint',
+            atol=tol,
+            rtol=tol
+        )
+        return res.y[-1]
+
+Simulate with low tolerance for comparison to high accuracy solution.
+
+.. jupyter-execute::
+
+    yf_low_tol = ode_sim(1., 1e-13)
+    np.linalg.norm(yf_low_tol - yf_dyson)
+
+
+For speed comparison, compile at a tolerance with similar accuracy.
+
+.. jupyter-execute::
+
+    jit_ode_sim = jit(lambda amp: ode_sim(amp, 1e-8))
+
+    %time yf_ode = jit_ode_sim(1.).block_until_ready()
+
+Measure compiled time.
+
+.. jupyter-execute::
+
+    %time yf_ode = jit_ode_sim(1.).block_until_ready()
+
+
+Confirm similar accuracy solution.
+
+.. jupyter-execute::
+
+    np.linalg.norm(yf_low_tol - yf_ode)
+
+Here we see that, once compiled, the Dyson-based solver has a
+significant speed advantage over the traditional solver, at the expense
+of the initial compilation time and the technical aspect of using the solver.
+
+5. How-to construct and simulate using the Magnus-based perturbation solver
+---------------------------------------------------------------------------
+
+Next, build the Magnus-based perturbative solver. The :class:`.MagnusSolver` uses the
```

Comment:
```suggestion
We repeat our example using the Magnus-based perturbative solver. Setup of the :class:`.MagnusSolver`
is the same to the :class:`.DysonSolver`, but uses the Magnus expansion and
matrix exponentiation to simulate over each fixed time step.
```
Im not sure if it should be *the same as * or *similar to*?

Reply:
Put this in - I think "similar to" is good here.